fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -794,62 +774,36 @@ private SqlQueryData processDateParm(Class<?> resourceType, Parameter queryParm,
             }
 
             datetime = QueryBuilderUtil.getInstant(value.getValueDate());
-            // If the dateTime value is fully specified down to the microsecond, build a where clause segment with strict equals.
-            if (!value.getValueDate().isPartial() && QueryBuilderUtil.hasSubSeconds(value.getValueDate().getValue())) {
-                start = datetime;
-                end = datetime;
-                if (isDateSearch) {
-                    whereClauseSegment.append(LEFT_PAREN);
-                    whereClauseSegment.append(tableAlias + DOT).append(DATE_VALUE)
-                                      .append(operator.value())
-                                      .append(BIND_VAR);
-                    bindVariables.add(Timestamp.from(datetime));
-                    whereClauseSegment.append(RIGHT_PAREN);
-                }
+            // If the search parameter value includes fractional seconds and the target value is an instant, then use equals
+            if (!value.getValueDate().isPartial() && QueryBuilderUtil.hasSubSeconds(value.getValueDate().getValue()) && isInstantSearch) {
+                whereClauseSegment.append(LEFT_PAREN);
+                whereClauseSegment.append(tableAlias + DOT).append(DATE_VALUE)
+                                  .append(operator.value())
+                                  .append(BIND_VAR);
+                bindVariables.add(Timestamp.from(datetime));
+                whereClauseSegment.append(RIGHT_PAREN);
+                whereClauseSegment.append(RIGHT_PAREN);
             } else {
```

Comment:
i think this whole logic branch can get deleted if we add that "inclusive" boolean into the db and handle all datetime searches (except `_lastUpdated`) through DATE_START and DATE_END